### ISSUE TITLE
incorrect handling of null values

### ISSUE THREAD

**Issue by anthonygreen** [appium/appium#5910]
It seems there maybe several places in the appium code where null values are handled incorrectly

example Safari fails to launch because

```
Unable to find app dir /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulatornull.sdk/Applications
info: [debug] Trying app dir /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS null.simruntime/Contents/Resources/RuntimeRoot/Applications
warn: App is also not at /tmp/Appium-MobileSafari-null.app
```

Clearly null here should be the Simulator version number or default to an empty string
Ditto iOS null.simruntime and Appium-MobileSafari-null.app

* OSX 10.10.5
* Xcode 7.1
* appium 1.4.16

**Comment by Jonahss** [comment]
Issues of this sort should be addressed by Appium version 1.5. Checkout the `1.5` branch to help us beta test ^.^

**Comment by pr4bh4sh** [comment]
It's working with Appium 1.5.0-beta8+xcode 6.3 + iPhone 4s (8.3 simulator) :thumbsup:
